src/utils/arrays.ts

```diff
@@ -223,6 +223,21 @@ export function arrayMerge<T>(...a: T[][]): T[] {
     }, new Set<T>()));
 }
 
+/**
+ * Moves a single element from fromIndex to toIndex.
+ * @param list the list from which to construct the new list.
+ * @param fromIndex the index of the element to move.
+ * @param toIndex the index of where to put the element.
+ * @returns A new array with the requested value moved.
+ */
+export function reorder<T>(list: T[], fromIndex: number, toIndex: number): T[] {
```

Comment:
nitpick: `reposition`, `moveWithin`, etc - reordering almost has an expectation of taking a property name to reorder based on that.